uf-mil/mil
# Improvements to mission system

## Description

This PR brings the following improvements to missions:
* Renaming `NaviGator` class to `NaviGatorMission` to better identify the class as a base mission
* Renaming `SubjuGator` class to `SubjuGatorMission` for same reason
* Exceptions are now printed through the mission client, and the mission server (instead of just the server)
* `sub8_missions.exceptions.KilledException` was added to add fail-fast breaking when a user attempts to start a mission while the sub is killed - this exception is raised if a move is requested while the sub is killed
* Exception handling in mission server is simplified
* `mil_missions_core.SubmissionException` is used for presenting errors in missions rather than just printing out the traceback
* SubjuGator's move proxy now uses an async `go()` method - this throws the `KilledException` mentioned above if the sub is killed and a move is requested
* `mil_missions_core.BaseMission` methods and attributes were renamed to names that better identify point of methods/attributes
* All setup and shutdown methods are now presumed to be async by default, sync setup/shutdown methods are not supported by missions anymore


## Screenshot or Video


https://user-images.githubusercontent.com/52760912/205992267-f59d3175-28b2-4f4b-b395-da36277fc7ad.mp4

## Related Issues


None

## Testing

1. Launch a robot simulation (Sub8/Sub9/NaviGator)
2. Observe that no mission can be run while robot is killed
3. Observe that exceptions are printed to mission client

## About This PR
<!-- BELOW: Have you checked the following? -→

- [x] I have updated documentation related to this change so that future members are aware of the changes I've made.
